compiler/rustc_codegen_llvm/src/debuginfo/mod.rs

```diff
@@ -121,6 +121,7 @@ pub fn finalize(cx: &CodegenCx<'_, '_>) {
         // This can be overridden using --llvm-opts -dwarf-version,N.
         // Android has the same issue (#22398)
         if cx.sess().target.target.options.is_like_osx
+            || cx.sess().target.target.options.is_like_bsd
```

Comment:
```suggestion
            || cx.sess().target.target.target_os.ends_with("bsd")
```
I'd avoid introducing a new flag until there are at least couple more use cases for it, or until BSD targets with different naming conventions appear.
(E.g. I'd remove `is_like_android` as well since it's a duplicate of `target_os == "android"`.)
`target_os` should be ok for ad hoc checks like this until they turn into something systematic.

Reply:
>or until BSD targets with different naming conventions appear

Actually, isn't `compiler\rustc_target\src\spec\dragonfly_base.rs` a BSD as well?

Reply:
Ah right it is, I forgot that one. It's most similar to freebsd and it looks like its`target_os` value is `"dragonfly"`. 

Is that enough of a naming convention difference to warrant my `is_like_bsd` changes, or should I switch to using something like:
```
            || cx.sess().target.target.target_os.ends_with("bsd")
            || cx.sess().target.target.target_os.equals("dragonfly")
```

Reply:
Let's use `ends_with("bsd") || equals("dragonfly")` for now.

Reply:
Disregard the above, let's add a `dwarf_version: Option<u32>` to `TargetOptions` instead.
It will default to `None` and set to `Some(2)` for the targets listed here.